src/exotic-sizes.md

```diff
@@ -106,9 +106,9 @@ consequence of types with no size. In particular, pointer offsets are no-ops,
 and allocators typically [require a non-zero size][alloc].
 
 Note that references to ZSTs (including empty slices), just like all other
-references, must be non-null and suitably aligned. Dereferencing a null or
-unaligned pointer to a ZST is [undefined behavior][ub], just like for any other
-type.
+references, must be non-null and suitably aligned. However, loading or storing
+through a null pointer to a ZST is not [undefined behavior][ub], unlike
+pointers to other types.
```

Comment:
I think it would be less confusing to say something like "loading or storing through a null pointer is UB, unless the load/store has size 0". Let's talk about the common case first.

Reply:
The subject of this section is ZSTs, so "loading or storing through a null pointer to a ZST is not undefined behavior" is the information it's supplying, and "unlike pointers to other types" is an aside.

Reply:
Ah, fair.